ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -823,4 +827,125 @@ var _ = Context("Inside the default namespace", func() {
+			err := k8sClient.Create(ctx, rayCluster)
+			Expect(err).NotTo(HaveOccurred(), "Failed to create RayCluster")
+			Eventually(
+				getResourceFunc(ctx, client.ObjectKey{Name: rayCluster.Name, Namespace: namespace}, rayCluster),
+				time.Second*3, time.Millisecond*500).Should(BeNil(), "Should be able to see RayCluster: %v", rayCluster.Name)
+
+			By("Check the number of worker Pods")
+			numWorkerPods := 3
+			Eventually(
+				listResourceFunc(ctx, &workerPods, workerFilters...),
+				time.Second*3, time.Millisecond*500).Should(Equal(numWorkerPods), fmt.Sprintf("workerGroup %v", workerPods.Items))
+
+			By("Check RayCluster conditions empty initially")
+			Expect(testRayCluster.Status.Conditions).To(BeEmpty())
+
+			By("Get Head Pod")
+			err = k8sClient.List(ctx, &headPods, headFilters...)
+			Expect(err).NotTo(HaveOccurred(), "Failed to list head Pods")
+			Expect(headPods.Items).Should(HaveLen(1), "headPods: %v", headPods.Items)
+			headPod = headPods.Items[0]
+
+			By("Update Head pods to Running")
+			headPod.Status.Phase = corev1.PodRunning
+			headPod.Status.Conditions = []corev1.PodCondition{
+				{
+					Type:   corev1.PodReady,
+					Status: corev1.ConditionTrue,
+				},
+			}
+			Expect(k8sClient.Status().Update(ctx, &headPod)).Should(Succeed())
+
+			By("Check RayCluster HeadPodReady condition is true")
+			Eventually(
+				func() bool {
+					if err := getResourceFunc(ctx, client.ObjectKey{Name: rayCluster.Name, Namespace: namespace}, rayCluster)(); err != nil {
+						return false
+					}
+					return meta.IsStatusConditionPresentAndEqual(rayCluster.Status.Conditions, string(rayv1.HeadPodReady), metav1.ConditionTrue)
+				},
+				time.Second*3, time.Millisecond*500).Should(BeTrue())
+
+			By("Check RayCluster RayClusterReady condition is false")
+			Consistently(
+				func() *metav1.Condition {
+					if err := getResourceFunc(ctx, client.ObjectKey{Name: rayCluster.Name, Namespace: namespace}, rayCluster)(); err != nil {
+						return nil
+					}
+					return meta.FindStatusCondition(rayCluster.Status.Conditions, string(rayv1.RayClusterReady))
+				},
+				time.Second*3, time.Millisecond*500).Should(BeNil())
+
+			By("Update Worker pods to Running")
+			for _, workerPod := range workerPods.Items {
+				workerPod.Status.Phase = corev1.PodRunning
+				workerPod.Status.Conditions = []corev1.PodCondition{
+					{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionTrue,
+					},
+				}
+				Expect(k8sClient.Status().Update(ctx, &workerPod)).Should(Succeed())
+			}
+
+			By("Check RayCluster RayClusterReady condition is true")
+			Eventually(
+				func() bool {
+					if err := getResourceFunc(ctx, client.ObjectKey{Name: rayCluster.Name, Namespace: namespace}, rayCluster)(); err != nil {
+						return false
+					}
+					return meta.IsStatusConditionPresentAndEqual(rayCluster.Status.Conditions, string(rayv1.RayClusterReady), metav1.ConditionTrue)
+				},
+				time.Second*3, time.Millisecond*500).Should(BeTrue())
```

Comment:
Can you check the following cases:

1. Update a worker pod to become unready; the `RayClusterReady` status should still be true.
2. Update the head pod to become unready; the `RayClusterReady` status should be false.